presto-main/src/main/java/com/facebook/presto/cost/ScalarStatsCalculator.java

```diff
@@ -101,5 +105,54 @@ protected SymbolStatsEstimate visitLiteral(Literal node, Void context)
             }
             return estimate.build();
         }
+
+        protected SymbolStatsEstimate visitCast(Cast node, Void context)
+        {
+            SymbolStatsEstimate sourceStats = process(node.getExpression());
+            TypeSignature targetType = TypeSignature.parseTypeSignature(node.getType());
+
+            // todo - make this general postprocessing rule.
+            double distinctValuesCount = sourceStats.getDistinctValuesCount();
+            double lowValue = sourceStats.getLowValue();
+            double highValue = sourceStats.getHighValue();
+
+            if (isIntegralType(targetType)) {
+                if (isFinite(lowValue)) {
```

Comment:
I think round behaves correctly with `-Inf` and `Inf` double values. I think it's just enough to test for `NaN`